frontend/institution/allInstitutionsController.js

```diff
@@ -89,11 +89,66 @@
         /**
          * Start function.
          * Sets the scrollListener and retrieve the first 10
-         * institutions.
+         * institutions if the cache is not cleaned otherwise it calls
+         * setupInstitutions.
          */
         allInstitutionsCtrl.$onInit = function () {
-            allInstitutionsCtrl.loadMoreInstitutions();
+            localforage.length().then(size => {
+                if (size === 0) {
+                    allInstitutionsCtrl._setupSeenInstitutions();
+                } else {
+                    allInstitutionsCtrl.loadMoreInstitutions();
+                }
+            });
             Utils.setScrollListener(content, allInstitutionsCtrl.loadMoreInstitutions);
         };
+
+        /**
+         * @private
+         * Get all institutions and insert them into localforage. This method
+         * is called when the cache is cleaned. After that, it calls loadMoreInstitutions.
+         */
+        allInstitutionsCtrl._setupSeenInstitutions = function _setupSeenInstitutions() {
```

Comment:
I understand that, when you clen the cache, every institutions are being marked as new, thats right?
If it is, I think that its not necessary. better new not marked, than old marked.

Reply:
but there is no other way to do that, how would I know that an institution is new to put it in idb considering I don't have the old ones in another place?